ESH-INF/thing/_channels.xml

```diff
@@ -3,15 +3,31 @@
 	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
 	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
 
-	<!-- Color Channel -->
-	<channel-type id="color_color">
-		<item-type>Color</item-type>
-		<label>Color</label>
-		<description>The color channel allows to control the color of a light.
+    <!-- Color Channel -->
+    <channel-type id="color_color">
+        <item-type>Color</item-type>
+        <label>Color</label>
+        <description>The color channel allows to control the color of a light.
             It is also possible to dim values and switch the light on and off.
         </description>
-		<category>ColorLight</category>
-	</channel-type>
+        <category>ColorLight</category>
+    </channel-type>
+
+    <!-- Color Channel -->
+    <channel-type id="color_mode">
+        <item-type>Number</item-type>
+        <label>Color</label>
+        <description>The color mode channel allows to control the color mode of a light.
+        </description>
+        <category>ColorLight</category>
+        <state readOnly="true">
+            <options>
+                <option value="0">Hue and Saturation</option>
+                <option value="1">X and Y</option>
```

Comment:
How do you send X/Y commands at all?

Reply:
At the moment it's not supported, but it will soon be added. The binding will decide based on what the bulb supports which system to use. (some) Tradfri uses XY, and Hue uses HS. The binding will discover what the bulb supports and use the appropriate system.

Reply:
> but it will soon be added

No, please don't! Checkout the Tradfri binding: Sure, the device wants XY, but this conversion is done by the binding - the channel is still proper HSB and nothing else. The same must be done here, otherwise you completely ignore the ESH architecture.

Reply:
Yes- agreed - the binding does the change. This is 100% for sure. IT will expose a standard HSB - no problem. BUT, the device needs to use XY - it doesn't work otherwise. The binding changes the HSB into XY and sends it to the bulb. The bulb will still report XY in its color mode though - I can't change that (unless we just remove the mode status channel).

Sorry - I clearly miss something, or maybe you don't understand what this mode channel is? It's ONLY a status channel to indicate what the BULB is doing - not ESH. ESH, and the binding, will always expose the standard channel types.

Reply:
The binding is the only one who needs to know what the bulb is doing - therefore you can remove the channel.
As I said, it is perfectly fine for the binding itself to internally read and manage that information, so that it knows, how to communicate with the bulb. But this must not be exposed to anyone else through an ESH channel.

Reply:
Ok, I added the channel at user request (@AfromD) - maybe he can explain why it's useful?